recipe/meta.yaml

```diff
@@ -1,311 +1,309 @@
-  skip: true  # [win32 or s390x]
+  number: {{ build }}
+  # Package not supported on s390x
+  skip: true  # [s390x]
+  string: headless_py{{ PY_VER_MAJOR }}{{ PY_VER_MINOR }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}            # [qt_version == "none"]
+  string: qt{{ qt_version }}_py{{ PY_VER_MAJOR }}{{ PY_VER_MINOR }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [qt_version != "none"]
+  run_exports:
+    # https://abi-laboratory.pro/index.php?view=timeline&l=opencv
+    # Things seem to change every patch versions, mostly the dnn module
+    - {{ pin_subpackage('libopencv', max_pin='x.x.x') }}
   ignore_run_exports_from:
     - python
-  missing_dso_whitelist:     # [osx]
-    - "*"   # [osx]
 
 requirements:
   build:
+    # pkg-config is required to find ffpmeg
+    - pkg-config
+    - cmake
+    - ninja
+    - libgomp                        # [linux]
+    - m2-patch  # [win]
+    # ICE when enabling this
+    # - llvm-openmp                    # [osx]
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
-    - curl
-    - unzip     # [not win]
-    - ninja-base
-    - pkg-config
-    - {{ cdt('mesa-libGL-devel') }}  # [linux]
-    - {{ cdt('mesa-libEGL-devel') }}  # [linux and not ppc64le]
+    - {{ cdt('libselinux') }}        # [linux]
+    - {{ cdt('libxau-devel') }}      # [linux]
+    - {{ cdt('libxcb') }}            # [linux]
+    - {{ cdt('libxdamage') }}        # [linux]
+    - {{ cdt('libxdamage-devel') }}  # [linux]
+    - {{ cdt('libxext-devel') }}     # [linux]
+    - {{ cdt('libxfixes') }}         # [linux]
+    - {{ cdt('libxfixes-devel') }}   # [linux]
+    - {{ cdt('libxi-devel') }}       # [linux]
+    - {{ cdt('libxrender-devel') }}  # [linux]
+    - {{ cdt('libxxf86vm') }}        # [linux]
+    - {{ cdt('mesa-libgl-devel') }}  # [linux]
+    - {{ cdt('mesa-libegl-devel') }}  # [linux]
+    - {{ cdt('mesa-dri-drivers') }}  # [linux]
   host:
-    - cmake
-    - jasper 2.0.14            # [not s390x]
-    - eigen 3.3.7
-    - ffmpeg 4.2.2             # [not (linux and s390x)]
-    - freetype 2.10
-    - glib 2                   # [not ppc64le]
+    - python
+    - numpy
+    - eigen 3.4.0
+    # Specify lgpl version of ffmpeg so that there are
+    # no quesitons about the license of the resulting opencv
+    # binary
+    - ffmpeg
+    - freetype 2.12
+    # harfbuzz, glib, gettext are both needed for freetype support
+    - harfbuzz 4.3.0
+    - glib 2
     - gst-plugins-base 1.14    # [not (win or ppc64le)]
     - gstreamer 1.14           # [not (win or ppc64le)]
     - gstreamer 1.18.5         # [win]
     - gst-plugins-base 1.18.5  # [win]
-    - harfbuzz 4.3.0           # [not win]
     - hdf5 1.12.1
+    # Thw OpenCV maintainers says that OpenJPEG have higher priority than JasPer which is deprecated. 
+    # In order to use JasPer, OpenJPEG must be disabled.
+    # see https://github.com/opencv/opencv/blob/4.10.0/doc/tutorials/introduction/config_reference/config_reference.markdown#png-jpeg-tiff-webp-support
```

Comment:
Then is there a reason for using jasper over openjpeg for wi and arm64?

Reply:
I left `openjpeg` only